### PR TITLE
[formrecognizer] Pylint next fixes

### PR DESCRIPTION
**Pull request by catalinaperalta**
This PR fixes pylint-next errors. With these updates we're at a 10/10 rating with pylint-next.

NOTE: Typing on dicts will be further improved in subsequent typing PRs, Rohit also has a PR open that will help improve Dict typing.

### CODE REVIEW

File: sdk/formrecognizer/azure-ai-formrecognizer/azure/ai/formrecognizer/_models.py [hunk 1]
```diff
@@ -350,7 +358,11 @@ def __new__(cls, first_page_number: int, last_page_number: int) -> "FormPageRang
         )
 
     def to_dict(self) -> Dict:
-        """Returns a dict representation of FormPageRange."""
+        """Returns a dict representation of FormPageRange.
+
+        :return: Dict
+        :rtype: Dict
```
Comment (l0lawrence) [inline sdk/formrecognizer/azure-ai-formrecognizer/azure/ai/formrecognizer/_models.py:364]:
```suggestion
        :rtype: dict
```

Comment (l0lawrence) [reply sdk/formrecognizer/azure-ai-formrecognizer/azure/ai/formrecognizer/_models.py:364]: and below the same

**Review by l0lawrence — APPROVED** [review]
few small nits about typing in docstring, but looks good! Another good test is to see if sphinx passes when run on your docstrings

File: sdk/formrecognizer/azure-ai-formrecognizer/azure/ai/formrecognizer/_helpers.py [hunk 2]
```diff
@@ -75,17 +76,28 @@ def adjust_value_type(value_type):
     return value_type
 
 
-def adjust_confidence(score):
-    """Adjust confidence when not returned."""
+def adjust_confidence(score: Optional[float]) -> float:
+    """Adjust confidence when not returned.
+
+    :param Optional[float] score: Confidence score to be adjusted.
```
[the remark this answers is not in the document]
Comment (catalinaperalta) [reply sdk/formrecognizer/azure-ai-formrecognizer/azure/ai/formrecognizer/_helpers.py:82]: Is this having `Optional[float]` incorrect or is it against the convention? Just wanted to make sure I know the right thing to do :)

[the remark this answers is not in the document]
Comment (l0lawrence) [reply sdk/formrecognizer/azure-ai-formrecognizer/azure/ai/formrecognizer/_helpers.py:82]: I don't believe it is incorrect  as Sphinx can generate using it, but the goal is to not use typing formatting in the docstrings.

**Pull request merged**